File: hej/apps/hej-web/lib/api/disputes.ts
/**
 * Project-level dispute case read API.
 */
import { apiClient, type ApiResult } from "./client"

export type ApiDisputeCase = {
  id: string
  task_id: string
  task_item_id: string
  status: string
  severity: string
  opened_by: string
  assigned_to: string
  disagreement_summary: string
  created_at: string
}

export function getProjectDisputes(
  projectId: string,
): Promise<ApiResult<ApiDisputeCase[]>> {
  return apiClient.get<ApiDisputeCase[]>(`/projects/${projectId}/disputes`)
}

File: hej/apps/hej-api/app/schemas/disputes.py
"""Schemas for project-level dispute case visibility."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class DisputeCaseRead(BaseModel):
    id: str
    task_id: str
    task_item_id: str
    status: str
    severity: str
    opened_by: str
    assigned_to: str
    disagreement_summary: str
    created_at: datetime
